tests/test_invenio_communities.py

```diff
@@ -199,56 +199,25 @@ def test_oaipmh_sets(app):
         assert OAISet.query.count() == 0
 
 
-def test_communities_rest_all_communities(app):
+def test_communities_rest_all_communities(app, communities_creation):
     """Test the OAI-PMH Sets creation."""
-    with app.app_context():
-        # Create a user and some communities
-        user1 = create_test_user()
-        comm1 = Community(id='comm1', id_user=user1.id)
-        comm2 = Community(id='comm2', id_user=user1.id)
-        db.session.add(comm1)
-        db.session.add(comm2)
-        db.session.commit()
-
-        with app.test_client() as client:
-            response = client.get('/api/communities/')
-            response_data = json.loads(response.get_data(as_text=True))
-            assert response_data['hits'] == {
-                'hits': [
-                    {
-                        'last_record_accepted': (
-                            '2000-01-01T00:00:00+00:00'),
-                        'description': '',
-                        'title': '',
-                        'page': '',
-                        'id': 'comm1',
-                        'curation_policy': '',
-                        'links': {
-                            'self': ('http://inveniosoftware.org/'
-                                     'api/communities/comm1'),
-                            'html': ('http://inveniosoftware.org/'
-                                     'communities/comm1/'),
-                        },
-                    },
-                    {
-                        'last_record_accepted': (
-                            '2000-01-01T00:00:00+00:00'),
-                        'description': '',
-                        'title': '',
-                        'page': '',
-                        'id': 'comm2',
-                        'curation_policy': '',
-                        'links': {
-                            'self': ('http://inveniosoftware.org/'
-                                     'api/communities/comm2'),
-                            'html': ('http://inveniosoftware.org/'
-                                     'communities/comm2/'),
-                        },
-                    },
-                ],
-                'total': 2,
-            }
-            assert len(['links']) == 1
+    with app.test_client() as client:
+        response = client.get('/api/communities/')
+        response_data = json.loads(response.get_data(as_text=True))
+        assert response_data['hits']['total'] == 3
+        assert len(response_data['hits']['hits']) == 3
+
+        comm = response_data['hits']['hits'][0]
+        assert comm['description'] == 'Description1'
```

Comment:
you could build a function ``assert_community_serialization()`` which takes care of this to simplify the test.